idaes/generic_models/unit_models/heat_exchanger.py

```diff
@@ -287,10 +295,16 @@ def _process_config(self):
             # compatible with the tube and shell notation
             setattr(config, config.cold_side_name, config.cold_side_config)
 
-        if config.cold_side_name in ["hot_side", "side_1"]:
-            raise ConfigurationError("Cold side name cannot be in ['hot_side', 'side_1'].")
-        if config.hot_side_name in ["cold_side", "side_2"]:
-            raise ConfigurationError("Hot side name cannot be in ['cold_side', 'side_2'].")
+        if config.side_1_is_hot:
```

Comment:
This seems like additional complexity and a potential point of confusion for users in the future - I assume there is a reason we need to know the hot side explicitly at some point for the more complex models?

My first (perhaps naive) reaction is that we should either force the issue and just make one side hot_side (and drop the generic side_1, side_2 names) if if actually matters, or just not assign any actual meaning to the names at all (i.e. you can call it hot side if you want, but we won't treat it any differently). Alternatively, we could use a standard (internal at least) that e.g. side_1 will always be the hot side.

Reply:
Yeah, this is a bit of a mess.  It generally doesn't matter which side is hot or cold.  But designating one side as hot helps since it may matter sometimes and it give you a point of reference if say the heat transfer is negative.  I don't think it hurts for the user to specify which side they think is hot.  The problem with side_1 and side_2 is that, at least, I can't keep them straight.  Of course, you can always name the sides anything you want.

The complex business with the side_1 and side_2 is for backward compatibility.  Also the boiler heat exchanger was different than the standard heat exchanger so I had to add another option about whether side_1 was cold or hot.

Anyway by default the sides are named "shell" and "tube".  It's a mess so as not to break any old models.  I guess it's an IDAES 2.0 cleanup action.